sootup.core/src/main/java/sootup/core/validation/FieldRefValidator.java

```diff
@@ -35,31 +41,47 @@ public List<ValidationException> validate(Body body, View<?> view) {
 
     List<ValidationException> validationException = new ArrayList<>();
 
-    // TODO: check copied code from old soot
-    /*
-     * SootMethod methodRef = body.getMethod(); if (methodRef.isAbstract()) { return; }
-     *
-     * Chain<Unit> units = body.getUnits().getNonPatchingChain();
-     *
-     * for (Unit unit : units) { Stmt s = (Stmt) unit; if (!s.containsFieldRef()) { continue; } JFieldRef fr =
-     * s.getFieldRef();
-     *
-     * if (fr instanceof JStaticFieldRef) { JStaticFieldRef v = (JStaticFieldRef) fr; try { SootField field = v.getField();
-     * if (field == null) { exceptions.add(new UnitValidationException(unit, body, "Resolved field is null: " +
-     * fr.toString())); } else if (!field.isStatic() && !field.isPhantom()) { exceptions .add(new
-     * UnitValidationException(unit, body, "Trying to get a static field which is non-static: " + v)); } } catch
-     * (ResolutionFailedException e) { exceptions.add(new UnitValidationException(unit, body,
-     * "Trying to get a static field which is non-static: " + v)); } } else if (fr instanceof InstanceFieldRef) {
-     * InstanceFieldRef v = (InstanceFieldRef) fr;
-     *
-     * try { SootField field = v.getField(); if (field == null) { exceptions.add(new UnitValidationException(unit, body,
-     * "Resolved field is null: " + fr.toString())); } else if (field.isStatic() && !field.isPhantom()) { exceptions.add(new
-     * UnitValidationException(unit, body, "Trying to get an instance field which is static: " + v)); } } catch
-     * (ResolutionFailedException e) { exceptions.add(new UnitValidationException(unit, body,
-     * "Trying to get an instance field which is static: " + v)); } } else { throw new RuntimeException("unknown field ref");
-     * } }
-     */
+    List<Stmt> stmts = body.getStmts();
 
+    for (Stmt stmt : stmts) {
+      if (!stmt.containsFieldRef()) {
+        continue;
+      }
+      JFieldRef fr = stmt.getFieldRef();
+
+      if (fr instanceof JStaticFieldRef) {
+        JStaticFieldRef v = (JStaticFieldRef) fr;
+
+        Optional<? extends SootField> fieldOpt = view.getField(v.getFieldSignature());
+        if (!fieldOpt.isPresent()) {
+          validationException.add(new ValidationException(v, "Resolved field is empty: "));
```

Comment:
this can occur when not a complete program is given in the input - lets just ignore those cases.